### DISCUSSION TITLE
RFC - How to deal with basic auth

### DISCUSSION THREAD

**Discussion by Confectrician** [openhab/openhab-vscode#357] [Polls]
I am facing error reports with "not working" connections and several times the reason where configured basic auth credentials on installations without a basic auth setup.

People think the have to fill in ther connection credentials for their ssh acces or their main ui credentials.
I am looking for feedback on how we can improve this.

I would like to keep basic auth settings somehow, but make it kind of harder to configure.
So that people who know why they are configuring basic auth, have little more effort to do and users who just fill those fields "beccause they are available" get pointed on a possible misconfiguration.

Any ideas on how we can improve this are welcome.

**Comment by Confectrician** [comment]
Another user who has configured basicAUth with the main ui credentials.
https://community.openhab.org/t/rpi-4-high-load-with-openhab-extension-1-0-0/122126/10

We need some kind of clarification.

**Comment by pgfeller** [comment]
@Confectrician, @florian-h05:

I propose to drop basic authentication and only support API key. This reduces complexity and code without removing value for the users in my opinion. What do you think?

If you agree I'll plan to remove the basic authentication code completely from the extension.

**Comment by florian-h05** [comment]
I think the reason basic auth is supported is that when running behind a reverse proxy with basic auth (as described in the docs), you need it.
